StackExchange/Stacks
# chore(tag): update design

[STACKS-459](https://stackoverflow.atlassian.net/browse/STACKS-459)

This PR updates the design of the tag component. 

## To test

1. Go to the [Tag page on the Deploy Preview](https://deploy-preview-1717--stacks.netlify.app/product/components/tags/)
2. Verify that tags render as expected
    - **Note**: Use the mode switcher in the topbar menu to check different combinations of light/dark/high-contrast
3. **Optional**: Check out the [Post Summary page on the Deploy Preview](https://deploy-preview-1717--stacks.netlify.app/product/components/post-summary/) to ensure tags render as expected in this context

## It includes

- Updated styles for the default, moderator, and required tags
- Removal of the `muted` variant
- Removal of the `is-selected` modifier

## It _does not_ include

- Any "Teams" tag styles (those are handled in Core with `--theme-tag-*` CSS variables)
- Any changes to the post summary component
    - Opacity-related fixes
    - Dark mode yellow background color

## Follow up integration tasks

We'll need to make a few updates to Core to ensure things render as expects. These include:

- Update styles of the `.post-tag` class to match updated `.s-tag` styles
- Removal of any `--theme-tag-*` variables from SE themes (I think. We'll need to at least check to see if this is necessary)
- Update Team tag colors to match expectations

## Screenshot comparisons

| Variant | Before | After |
|-|-|-|
| default | ![image](https://github.com/StackExchange/Stacks/assets/647177/f4cf348d-db1d-474d-9ec4-356594a21e82) | ![image](https://github.com/StackExchange/Stacks/assets/647177/2c2b9728-1bad-4455-8bce-1c54a2732aff) |
| moderator | ![image](https://github.com/StackExchange/Stacks/assets/647177/bc1f1cc0-fff8-4919-8e4d-bde74afcefda) | ![image](https://github.com/StackExchange/Stacks/assets/647177/65edf57c-7f32-46e8-b39e-668a9a116cea) |
| required | ![image](https://github.com/StackExchange/Stacks/assets/647177/c6022933-5eeb-48f1-9e94-1c5b8a21dd4e) | ![image](https://github.com/StackExchange/Stacks/assets/647177/2f540d09-57be-482b-97f9-10b182e0e627) |

---

cc @ellenchanhy (I'm cc-ing you here but if you'd like to be an "official" reviewer, you can ask helpdesk to be included in the `StackExchange` team. Once that's done, I can add you to the Stacks repo as a contributor)
